pandapower/io_utils.py

```diff
@@ -746,6 +748,23 @@ def from_json(cls, json_string):
         d = json.loads(json_string, cls=PPJSONDecoder)
         return cls.from_dict(d)
 
+    def __eq__(self, other):
+        """
+        isinstance: so that we know we can use to_json()
+        comparing class name instead of class directly allows more flexibility,
+        e.g. when the class definition is moved to a different module.
+        There is still a risk that the implementation details of the methods can differ
+        if the classes are from different modules.
```

Comment:
So it could be that as long as both objects have the same function names they are considered equal, correct? Thus, let's say controller 1 has the function:
>**def control_step():**
>self.check_word = 'banana'

and controller 2 has the same function, however, with different output:
>**def control_step():**
>self.check_word = 'apple'
they are still considered equal, correct?

I agree that this is a risk. However, that was is before too, so I still like your implementation. Still we should keep it in mind.

Reply:
If the attributes "check_word" are different, then they will be different. 
e.g. at first, before running runpp, the attribute "check_word" is not set, then nets_equal will be True.
After pp.runpp, the attributes will be set differently, and then it will return False. 
But only as long as the class is the same, e.g. due to a change in the implementation or because they were defined in different modules.

Reply:
But then if they are from different modules, DeepDiff will return False. I can change it so that the status (__dict__) is used to compare, and not the full object.

Reply:
Is it so? Did you try with the test in test_file_io where this happens?

Reply:
I changed it to .__dict__ comparison